CMakeLists.txt

```diff
@@ -10,7 +10,7 @@
 ###############################################################################
 # Minimum CMake version and policies
 ###############################################################################
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.1.0 FATAL_ERROR)
```

Comment:
I think this minimum version is too new. Our CI uses Ubuntu 14.04 LTS and that ships with [CMake 2.8.12](https://packages.ubuntu.com/trusty/cmake). You may have to revert to using the manual method of passing `-std=c++11` to the compiler.

Of course we could change the CI to use a newer version of CMake but this will also break the docker image because that also uses Ubuntu 14.04LTS as its base.